src/mapGenerator/mapgenManager.py

```diff
@@ -0,0 +1,156 @@
+# system imports
+import logging
+import sys
+from PyQt5 import QtCore, QtGui, QtWidgets, QtNetwork
+import os
+import re
+import shutil
+import random
+
+# local imports
+from config import Settings
+import fafpath
+from vault.dialogs import downloadFile
+from mapGenerator.mapgenProcess import MapGeneratorProcess
+from fa.maps import getUserMapsFolder
+
+logger = logging.getLogger(__name__)
+
+class MapGeneratorManager(object):
+    def __init__(self):
+        self.releaseUrl = "https://github.com/FAForever/Neroxis-Map-Generator/releases/download/{}/NeroxisGen_{}.jar"
+        self.latestRelease = "https://github.com/FAForever/Neroxis-Map-Generator/releases/latest"
+        self.latestVersion = None
+        self.response = None
+
+        self.currentVersion = Settings.get('mapGenerator/version', "0", str)
+        self.previousMaps = Settings.get('mapGenerator/deleteMaps')
+        self.generatorPath = os.path.join(fafpath.get_libdir(), "map-generator")
+        self.mapsFolder = getUserMapsFolder()
+
+        if self.previousMaps:
+            self.deletePreviousMaps()
+
+    def generateMap(self, mapname):
+        dataString =  mapname.split("neroxis_map_generator",1)[1]
+        version = re.search(r'\_(.*?)\_',dataString).group(1)
+        seed = dataString.rsplit('_', 1)[1]
+
+        actualPath = self.versionController(version)
+
+        if actualPath:
+            auto = Settings.get('mapGenerator/autostart', default=False, type=bool)
+            if not auto:
+                msgbox = QtWidgets.QMessageBox()
+                msgbox.setWindowTitle("Generate map")
+                msgbox.setText("Seems that you don't have the map used this game. Do you want to generate it?<br/><b>" + mapname + "</b>")
```

Comment:
It looks like you don't have the map being used by this lobby. Do you want to generate it? (also, no internationalization in the py client?)

Reply:
Nope.

Reply:
Yeah, I know my engrish skills are epic, that's why I always try to copy important msgs from somewhere https://github.com/FAForever/client/blob/develop/src/fa/check.py#L35 :D